Riot/Modules/VoiceBroadcast/VoiceBroadcastSDK/VoiceBroadcastAggregator.swift

```diff
@@ -108,27 +119,45 @@ public class VoiceBroadcastAggregator {
             
             self.events.append(contentsOf: response.chunk)
             
-            
-            let eventTypes = [VoiceBroadcastSettings.eventType, kMXEventTypeStringRoomMessage]
+            let eventTypes = [VoiceBroadcastSettings.voiceBroadcastInfoContentKeyType, kMXEventTypeStringRoomMessage]
             self.referenceEventsListener = self.room.listen(toEventsOfTypes: eventTypes) { [weak self] event, direction, state in
-                // TODO: check sender id to block fake voice broadcast chunk
                 guard let self = self,
+                      event.sender == self.voiceBroadcastSenderId,
                       let relatedEventId = event.relatesTo?.eventId,
                       relatedEventId == self.voiceBroadcastStartEventId,
                       event.content[VoiceBroadcastSettings.voiceBroadcastContentKeyChunkType] != nil else {
                     return
                 }
                 
+                if let chunk = self.voiceBroadcastBuilder.buildChunk(event: event, mediaManager: self.session.mediaManager, voiceBroadcastStartEventId: self.voiceBroadcastStartEventId) {
+                    self.delegate?.voiceBroadcastAggregator(self, didReceiveChunk: chunk)
+                }
+                
                 self.events.append(event)
+                MXLog.debug("[VoiceBroadcastAggregator] Got a new chunk for broadcast \(relatedEventId). Total: \(self.events.count)")
                 
-                self.voiceBroadcast = self.voiceBroadcastBuilder.build(voiceBroadcastStartEventContent: self.voiceBroadcastInfoStartEventContent,
-                                                   events: self.events,
-                                                   currentUserIdentifier: self.session.myUserId)
+                self.voiceBroadcast = self.voiceBroadcastBuilder.build(mediaManager: self.session.mediaManager,
```

Comment:
I think this object "self.voiceBroadcast" should be updated instead of be replaced ->TODO